app/views/ladder/LadderView.coffee

```diff
@@ -71,19 +69,68 @@ module.exports = class LadderView extends RootView
 
     @displayTabContent = 'display: block'
 
+    @calcTimeOffset()
+    @mandate = @supermodel.loadModel(new Mandate()).model
+
     @loadLeague()
     @urls = require('core/urls')
 
-  checkForTournamentEnd: =>
-    return if @destroyed
-    return false if me.isAdmin()
-    $.get '/db/mandate', (data) =>
-      return if @destroyed
-      if data?[0]?.currentTournament isnt 'magic-rush'
-        @tournamentEnd = true
-        @displayTabContent = 'display: none'
+    if features.china
+      setTimeout =>
+        @checkTournamentEndInterval = setInterval @checkTournamentEnd(), 10000
+      , 1000
+
+  calcTimeOffset: ->
+    $.ajax
+      type: 'HEAD'
+      success: (result, status, xhr) =>
+        @timeOff = new Date(xhr.getResponseHeader("Date")).getTime() - Date.now()
+
+  checkTournamentEnd: =>
+    return unless @timeOff
+    return unless @mandate.loaded
+    return unless @level.loaded
+    return if (@leagueID and not @league.loaded)
+    mandate = @mandate.get('0')
+
+    tournamentState = 1
+    #        tournamentState table
+    #  ladder\checkTournamentEnd   keep    stop
+    #  open                         3       2
+    #  close                        1       0
+    if mandate
+      tournamentState = @getTournamentState mandate, @courseInstance?.id, @level.get('slug'), @timeOff
+      if tournamentState > 1
+        if @tournamentEnd
+          @tournamentEnd = false
+          @render()
       else
-        setTimeout @checkForTournamentEnd, 60 * 1000
+        unless @tournamentEnd or me.isAdmin()
+            @tournamentEnd = true
+            @render()
+    if tournamentState % 2 == 0
+      clearInterval @checkTournamentEndInterval
+    return @checkTournamentEnd
+
+  getTournamentState: (mandate, courseInstanceID, levelSlug, timeOff) ->
```

Comment:
This is excellent but the returned values are hard to understand especially as they flow through the code.
When I am reading other places in the code I need to refer to what a tournament state of `1` or `3` means.
Instead could we assign the number to a constant?

Instead of code that looks like this:
```
if (tournamentState === 1)
  # Do something....
```

it would be better as:
```
# At top of file
TOURNAMENT_ENDED = 2

# In logic
if (tournamentState == TOURNAMENT_ENDED)
  # Do stuff...
```
^ This is more readable in the future. Could we make all the integers constants with appropriate names. Thank you!